Sources/Mistica/Components/Sheet/SheetTransition/SheetPresentationController.swift

```diff
@@ -0,0 +1,288 @@
+
+        let fittingSize = CGSize(width: containerView.bounds.width, height: UIView.layoutFittingCompressedSize.height)
+        let presentedViewHeight = presentedView.systemLayoutSizeFitting(
+            fittingSize,
+            withHorizontalFittingPriority: .required,
+            verticalFittingPriority: .defaultLow
+        ).height
+
+        /// The target height of the presented view.
+        /// If the size of the of the presented view could not be computed, meaning its equal to zero, we default to the maximum height.
+        let targetHeight = presentedViewHeight == .zero ? maximumHeight : presentedViewHeight
+
+        // Adjust the height of the view by adding the bottom safe area inset.
+        let adjustedHeight = min(targetHeight, maximumHeight) + containerView.safeAreaInsets.bottom
+
+        let targetSize = CGSize(width: containerView.frame.width, height: adjustedHeight)
+        let targetOrigin = CGPoint(x: .zero, y: containerView.frame.maxY - targetSize.height)
+
+        return CGRect(origin: targetOrigin, size: targetSize)
+    }
+
+    /// The `UIScrollView` embedded in the `presentedView`.
+    /// When available, the drag of the scroll view will be used to drive the interactive dismiss transition.
+    private var presentedScrollView: UIScrollView? {
+        guard let presentedView = presentedView else {
+            return nil
+        }
+
+        if let scrollView = presentedView.subviews.first(where: { $0 is UIScrollView }) as? UIScrollView {
+            return scrollView
+        }
+
+        return nil
+    }
+
+    /// The object that is managing the presentation and transition.
+    private var transitioningDelegate: SheetTransitioningDelegate? {
+        presentedViewController.transitioningDelegate as? SheetTransitioningDelegate
+    }
+
+    // MARK: - UI Elements
+
+    /// The view displayed behind the presented controller.
+    private lazy var overlayView: UIView = {
+        let view = UIView()
+        view.alpha = .zero
+        view.backgroundColor = .backgroundOverlay
+        view.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(tappedBackgroundView)))
+        return view
+    }()
+
+    /// The view displaying a handle on the presented view.
+    private let handleView: UIView = {
+        let view = UIView()
+        view.backgroundColor = .control
+        view.frame.size = CGSize(width: 40, height: 4)
+        return view
+    }()
+
+    /// The pan gesture used to drag and interactively dismiss the sheet.
+    private lazy var panGesture = UIPanGestureRecognizer(target: self, action: #selector(pannedPresentedView))
+
+    // MARK: - Methods
+
+    override func presentationTransitionWillBegin() {
+        super.presentationTransitionWillBegin()
+
+        containerView?.addSubview(overlayView)
+
+        presentedView?.addSubview(handleView)
+
+        overlayView.fade(toAlpha: 0.5, duration: 0.25)
```

Comment:
```suggestion
        overlayView.fade(toAlpha: 1, duration: 0.25)
```

`backgroundOverlay` has its own opacity I think you are applying opacity over the `backgroundOverlay` opacity, I think this value has to be `1` (or maybe remove?)

https://github.com/Telefonica/mistica-ios/blob/4d64130039f83128b8305012ce35575294c720a6/Sources/MisticaCommon/Colors/MovistarColorPalette.swift#L19

Reply:
Yes, you're right, fixed!